### PR TITLE
doc: Add instructions about using the external flash memory

### PR DESCRIPTION
**Pull request by divipillai**
The instruction to use the external flash memory is added to following applicaiton and samples:

Require external flash:
nrf/samples/nrf9160/modem_trace_flash_download
nrf/samples/nrf9160/modem_trace_flash
applications/asset_tracker_v2
 
Have optional features requiring external flash:
nrf/samples/nrf9160/modem_shell
nrf/samples/nrf9160/nrf_cloud_rest_fota
nrf/samples/nrf9160/nrf_cloud_mqtt_multi_service

CIA-855

### CODE REVIEW

File: samples/nrf9160/nrf_cloud_rest_fota/README.rst
Comment (divipillai) [inline samples/nrf9160/nrf_cloud_rest_fota/README.rst:21]: The content was moved to avoid back to back notes in the documentation.

File: doc/nrf/includes/external_flash_nrf91.txt
Comment (tokangas) [inline doc/nrf/includes/external_flash_nrf91.txt:1]: This text usually follows the table of supported platforms. If the table contains both Thingy:91 and nRF9160 DK, it's not clear that the above text refers to nRF9160 DK only.

File: applications/asset_tracker_v2/doc/asset_tracker_v2_description.rst
Comment (trantanen) [reply applications/asset_tracker_v2/doc/asset_tracker_v2_description.rst:23]: At least align all samples similarly in terms of TFM and external flash documentation order.

File: samples/nrf9160/modem_shell/README.rst
Comment (trantanen) [inline samples/nrf9160/modem_shell/README.rst:19]: I'll comment here but applies to all Requirements sections.
When looking at the generated document, it's starting to look like there could be subsections or something to distinguish between supported development kits, external flash, and TFM.

File: doc/nrf/working_with_nrf/nrf91/nrf9160.rst
Comment (gregersrygg) [inline doc/nrf/working_with_nrf/nrf91/nrf9160.rst:33]: Shouldn't «Program» have a lowercase «p»?

File: doc/nrf/working_with_nrf/nrf91/nrf9160.rst
Comment (gregersrygg) [inline doc/nrf/working_with_nrf/nrf91/nrf9160.rst:35]: The v0.9.0 board and earlier didn't have external flash, so this line can be removed.

**Review by gregersrygg — APPROVED** [review]

File: doc/nrf/includes/external_flash_nrf91.txt
Comment (jayteemo) [inline doc/nrf/includes/external_flash_nrf91.txt:5]: same as https://github.com/nrfconnect/sdk-nrf/pull/11008/files#r1183628876.

File: doc/nrf/includes/external_flash_nrf91.txt
Comment (jayteemo) [inline doc/nrf/includes/external_flash_nrf91.txt:1]: perhaps also mention somewhere the minimum DK version that has external flash... i think v0.14.0

**Review by jayteemo — APPROVED** [review]

File: applications/asset_tracker_v2/doc/asset_tracker_v2_description.rst
Comment (divipillai) [reply applications/asset_tracker_v2/doc/asset_tracker_v2_description.rst]: I have moved the external flash content after TFM.

File: samples/nrf9160/modem_shell/README.rst
Comment (divipillai) [reply samples/nrf9160/modem_shell/README.rst]: I believe paragraph break after each requirements (development kits, external flash, and TFM) and now moving the external flash content after TFM, makes the document more readable.

File: doc/nrf/working_with_nrf/nrf91/nrf9160.rst
Comment (divipillai) [reply doc/nrf/working_with_nrf/nrf91/nrf9160.rst]: Updated.

File: doc/nrf/includes/external_flash_nrf91.txt
Comment (divipillai) [reply doc/nrf/includes/external_flash_nrf91.txt]: Updated the "external_flash_nrf91.txt" with nRF9160 DK and the supported versions.

File: samples/nrf9160/modem_shell/README.rst
Comment (trantanen) [reply samples/nrf9160/modem_shell/README.rst]: The problem with paragraph breaks is that you'll have similar breaks for the note and numbered list.

Anyway, I'll leave this to the documentation team to decide. Maybe something you can discuss with your colleagues and address with separate PRs if needed.

**Review by trantanen — APPROVED** [review]

**Comment by trantanen** [comment]
@tokangas is out on Friday so feel free to get this merged without his approval. I checked that his comment was addressed.

File: samples/nrf9160/modem_shell/README.rst
Comment (trantanen) [reply samples/nrf9160/modem_shell/README.rst]: Your plan sounds good.

**Review by simensrostad — APPROVED** [review]

**Review by tokangas — APPROVED** [review]

**Review by plskeggs — APPROVED** [review]

**Pull request merged**
